docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,198 @@ Example for the `__time` dimension:
   "function" : "function(t) { return 'Second ' + Math.floor((t % 60000) / 1000); }"
 }
 ```
+
+### Namespaced extraction function (EXPERIMENTAL)
+A namespaced extraction function is a simple key-value mapping where the key-value mappings are unique to a particular namespace.
+This can be used for re-naming values in a datasource for certain queries. To use a particular namespace, simply set the appropriate namespace for the dimExtractionFn
+```json
+    "dimExtractionFn" : {
+      "type":"namespace",
+      "namespace":"some_namespace",
+      "missingValue":"NA",
+      "uniqueRenames":false
+    }
+```
+
+A property of `missingValue` can be specified at query time to hint how to handle missing values. Setting `missingValue` to `""` or `null` or omitting the property will cause the dimension's original value to be used.
```

Comment:
Good question. My initial thought was that the user should map it to *something* and handle the missing values on client side. If supporting mapping to null is required, then I suspect that may exacerbate the need for more unified null handling across null queries (instead of just groupBy where it is most commonly encountered now).